ui/src/ui-utils.ts

```diff
@@ -64,3 +64,62 @@ export function truncateText(text: string, limit: number, withEllipsis: boolean
     return text;
   }
 }
+
+/**
+ * Get the correct display name for certain publishers.
+ * 
+ * @param {string} the type of primaryPaperLInk
+ */
+export function getLinkText(input: string): string {
```

Comment:
nit pick: can we use a more descriptive name than `input` here? Perhaps `linkType`?